azurelinuxagent/common/cgroupapi.py

```diff
@@ -356,64 +364,28 @@ def start_extension_command(self, extension_name, command, timeout, shell, cwd,
 
         # Wait for process completion or timeout
         try:
-            try:
-                process_output = handle_process_completion(process=process,
-                                                           command=command,
-                                                           timeout=timeout,
-                                                           stdout=stdout,
-                                                           stderr=stderr,
-                                                           error_code=error_code)
-            finally:
-                with self._systemd_run_commands_lock:
-                    self._systemd_run_commands.remove(process.pid)
+            return handle_process_completion(process=process, command=command, timeout=timeout, stdout=stdout, stderr=stderr, error_code=error_code)
         except ExtensionError as e:
             # The extension didn't terminate successfully. Determine whether it was due to systemd errors or
             # extension errors.
-            systemd_failure = self._is_systemd_failure(scope, stderr)
-            process_output = read_output(stdout, stderr)
-
-            if not systemd_failure:
+            if not self._is_systemd_failure(scope, stderr):
                 # There was an extension error; it either timed out or returned a non-zero exit code. Re-raise the error
                 raise
+
+            # There was an issue with systemd-run. We need to log it and retry the extension without systemd.
+            process_output = read_output(stdout, stderr)
+            # Reset the stdout and stderr
+            stdout.truncate(0)
+            stderr.truncate(0)
+
+            if isinstance(e, ExtensionOperationError):
+                err_msg = 'Systemd process exited with code %s and output %s' % (e.exit_code, process_output)  # pylint: disable=no-member
```

Comment:
NIT: Unhandled Pylint error

Reply:
this code was moved here from elsewhere... I'll add a comment